base/applications/rapps/appview.cpp

```diff
@@ -1697,9 +1697,44 @@ CApplicationView::SetRedraw(BOOL bRedraw)
 }
 
 void
-CApplicationView::SetFocusOnSearchBar()
+CApplicationView::SetFocusOnSearchBar(UINT SetSel)
 {
     m_SearchBar->SetFocus();
+    if (SetSel)
+    {
+        m_SearchBar->SendMessage(EM_SETSEL, (INT8)LOBYTE(LOWORD(SetSel)), (INT8)HIBYTE(LOWORD(SetSel)));
+        m_SearchBar->SendMessage(EM_SETSEL, (INT8)LOBYTE(HIWORD(SetSel)), (INT8)HIBYTE(HIWORD(SetSel)));
```

Comment:
Why the explicit INT8 cast?

Reply:
I want sign extension

Reply:
Two separate parameters are preferred.

Reply:
> Two separate parameters are preferred

Don't you mean 4 parameters? I'm packing in 4 parameters here but each of them are only ever going to be 0 or -1.

You probably have to look up the documentation for EM_SETSEL anyway because nobody remembers the combination to set the cursor at the end of the string...

Reply:
Ok.

Reply:
> Don't you mean 4 parameters? I'm packing in 4 parameters here but each of them are only ever going to be 0 or -1.

As you can see, doing the packing makes the code much harder to understand for anybody else (except you who wrote it, of course).

Reply:
Separate question: Why does this need _two consecutive_ `EM_SETSEL` calls?
Normally, to select a range of characters, only one call is sufficient.

Reply:
> As you can see, doing the packing makes the code much harder to understand for anybody else (except you who wrote it, of course).

Ah yes, because `SetFocusOnSearchBar(0, -1, -1, 0)` is so much more readable.

I also have to look up EM_SETSEL on MSDN when I call it.

Leaving the selection active might be suitable in some situations so I let the caller specify exactly what they want.

I can maybe come up with a named constant you can pass.

> Why does this need _two consecutive_ `EM_SETSEL` calls?

To put the cursor at the end, you need to first select all the text and then remove the selection again.

Reply:
> To put the cursor at the end, you need to first select all the text and then remove the selection again.

OK that's the code comment that was missing in the function, please add this.

Which leads me to my next question: Was the aim of the single packed parameter of your `SetFocusOnSearchBar` to specify where to put the text cursor (so, zero: beginning; -1: end)?
If this is so, it appears our code base seems to have a simpler solution:
`SendMessage(EM_SETSEL, 0, 0);` would place the cursor at the beginning;

Reply:
Do you think I added this for fun? The caret blinks at the start of the text if I do nothing.

Reply:
On Windows? Weird.

Reply:
I did not test this on Windows. Typing in the Edit box will move the caret as expected but we are not typing here, we are just setting the window text directly.

Reply:
Yes I understand the text is already in the edit box for sure. The EM_SETSEL in the examples I linked to should not depend on whether you type in the box or something else...

Reply:
I need two messages, don't ask me why.

![screenshot](https://github.com/reactos/reactos/assets/138629473/23b1e539-9445-4aad-b6a0-2838335fd906)